src/OpenLoco/Map/MapGenerator.cpp

```diff
@@ -410,9 +411,20 @@ namespace OpenLoco::Map::MapGenerator
     // 0x004C4BD7
     static void generateWater(HeightMap& heightMap)
     {
-        _heightMap = heightMap.data();
-        call(0x004C4BD7);
-        _heightMap = nullptr;
+        static loco_global<uint16_t, 0x00525FB2> seaLevel;
+
+        Map::tile_loop tileLoop;
+        for (uint32_t posId = 0; posId < map_size; posId++)
+        {
+            auto pos = tileLoop.current();
+            auto tile = TileManager::get(pos);
+            auto* surface = tile.surface();
+
+            if (surface != nullptr && surface->baseZ() < (seaLevel << 2))
+                surface->setWater(seaLevel);
```

Comment:
It isn't obvious why we set water to sea level only if it is below sea level * 4

Reply:
I imagine it's to prevent setting water at surface level, which I imagine could lead to glitches on slopes and such. This is just a guess; for now it's just a re-implementation of what the base game is doing at the moment.